python/res/enkf/jobs/scaling/measured_data.py

```diff
@@ -0,0 +1,175 @@
+        for key in observation_keys:
+            observation_type = ert.getObservations()[key].getImplementationType().name
+            if observation_type == "GEN_OBS":
+                measured_data = measured_data.append(self._get_general_data(ert, key, case_name))
+            elif observation_type == "SUMMARY_OBS":
+                measured_data = measured_data.append(self._get_summary_data(ert, key, case_name))
+            elif observation_type == "BLOCK_OBS":
+                measured_data = measured_data.append(self._get_block_data(
+                    ert.getObservations(), key, ert.getEnsembleSize(), ert.getEnkfFsManager().getCurrentFileSystem())
+                )
+            else:
+                raise TypeError("Unknown observation type: {}".format(observation_type))
+        return measured_data
+
+    def _get_block_data(self, observations, key, ensamble_size, storage):
+        obs_vector = observations[key]
+        loader = PlotBlockDataLoader(obs_vector)
+
+        data = pd.DataFrame()
+        for report_step in obs_vector.getStepList().asList():
+
+            block_data = loader.load(storage, report_step)
+            obs_block = loader.getBlockObservation(report_step)
+
+            data = (
+                data
+                .append(pd.DataFrame(
+                    [self._get_block_observations(obs_block.getValue, obs_block)], index=["OBS_" + key]))
+                .append(pd.DataFrame(
+                    [self._get_block_observations(obs_block.getStd, obs_block)], index=["STD_" + key]))
+                .append(self._get_block_measured(key, ensamble_size, block_data))
+            )
+        return data
+
+    def _get_general_data(self, ert, observation_key, case_name):
+        ert_obs = ert.getObservations()
+        data_key = ert_obs[observation_key].getDataKey()
+
+        general_data = pd.DataFrame()
+
+        for time_step in ert_obs[observation_key].getStepList().asList():
+            data = GenDataCollector.loadGenData(ert, case_name, data_key, time_step)
+
+            general_data = (
+               general_data
+               .append(pd.DataFrame(self._get_observations(ert_obs, observation_key), index=["OBS_" + observation_key]))
+               .append(pd.DataFrame(self._get_std(ert_obs, observation_key), index=["STD_" + observation_key]))
+               .append(pd.concat([pd.DataFrame([data[key]], index=[observation_key]) for key in data.keys()]))
+            )
+        return general_data
+
+    def _filter_on_column_index(self, index_list):
+        """
+        Retuns a subset where the columns in index_list are filtered out
+        """
+        if isinstance(index_list, (list, tuple)):
+            if max(index_list) > self.data.shape[1]:
+                msg = ("Index list is larger than observation data, please check input, "
+                       "max index list: {} number of data points: {}".format)
```

Comment:
This seems like an expression to me. Not a statement since `format` does not seem to be called?

Reply:
Good catch, expanded `format` to give better user feedback.